### PR TITLE
🎖️ Leaderboard: hit the API 🎖️

### PR DESCRIPTION
**Pull request by kambereBr**
# 🏅 Features Implemented

Project Requirements: https://github.com/microverseinc/curriculum-javascript/blob/main/leaderboard/m2_send_receive_data.md

## 📝 General requirements

- ✅ No Linters error
- ✅ Used GitFlow.
- ✅ Used webpack to bundle JavaScript.
- ✅ Updated `README` file.
- ✅ Used ES6 modules, with import and export
- ✅ Send and receive data from an API.
- ✅ Used API documentation.
- ✅ Used JSON.
- ✅ Used JavaScript code asynchronous.
- ✅ New game created
- ✅ Some Players and their scores added to the game

### CODE REVIEW

File: src/modules/api.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+// Get All Score
+const getAllScore = async (htmlElement) => {
+  await fetch(SCORE_END_POINT)
+    .then((response) => response.json())
+    .then((json) => {
+      renderScore(json.result, htmlElement);
+    });
+};
```
Comment (brainconnect93) [inline src/modules/api.js:24]: - [ ] Great effort working on the `HIT the Api` milestone, but according to the project requirement, it was stated that when the refresh button is click that is when the data should be Gotten from the Api you submitted. But currently in your project the data is rendered on the list the moment the submit button is clicked. I would kindly suggest you work on it and be sure you follow the project requirement as much as possible to have a perfect projec.

Project requirement 👇 

image

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @kambereBr, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 

- Descriptive Pull Request title and summary. ✔️
- usage of `arrow functions` instead of the `function` keyword ✔️
- Use `async` and `await` with the `fetch` method ✔️
- Good use of Git flow ✔️ 


### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/api.js (hunk 1 above)
Comment (kambereBr) [reply src/modules/api.js:24]: Thanks dear @brainconnect93 for your review. I've updated my code according to your directions.

**Review by sumairq — APPROVED** [review]
### Hi @kambereBr , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kambereBr** [comment]
Thanks dear @sumairq

**Pull request merged**